### PR TITLE
Rover: clean up defines

### PR DESCRIPTION
**Pull request by peterbarker**
Mostly remove unused defines, but start using SPEED_CRUISE which looks like it should have been.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
LGTM.

File: APMrover2/defines.h [hunk 1]
```diff
@@ -10,19 +10,12 @@
 #define DEBUG 0
 #define SERVO_MAX 4500	// This value represents 45 degrees and is just an arbitrary representation of servo max travel.
 
-// active altitude sensor
-// ----------------------
-#define SONAR 0
```
Comment (gmorph) [inline APMrover2/defines.h]: I think this is required for SONAR logging yes?  The comment is clearly wrong.

Comment (peterbarker) [inline APMrover2/defines.h]: On Tue, 16 Aug 2016, Grant Morphett wrote:

> > -// active altitude sensor
> > -// ----------------------
> > -#define SONAR 0
> 
> I think this is required for SONAR logging yes? The comment is clearly
> wrong.

I don't think the define is used for logging.  "PLOG" is a nasty #define 
which mangles its arguments to form a different identifier.

The comment is also wrong :-)

**Comment by gmorph** [comment]
Applied - thanks!

**Pull request closed without merging**